src/js/lib/monster.util.js

```diff
@@ -1051,60 +1051,45 @@ define(function(require) {
 	}
 
 	/**
-	 * Helper function that takes an array of number in parameter, sorts it, and returns the first
-	 * number not in the array, greater than the minVal.
-	 * @param  {String[]} listNumbers Values treated as existing extensions.
-	 * @return {Number}             Next extension.
-	 */
-	function getNextExtension(listNumbers) {
-		var orderedArray = listNumbers;
-		var previousIterationNumber;
+	 * Finds the smallest number not in `list`, greater or equal to 1000.
+	 * @param  {String[]} list Values treated as existing numbers.
+	 * @return {Number}      Next number.
+	 */
+	function getNextExtension(list) {
 		var minNumber = 1000;
-		var lowestNumber = minNumber;
-		var increment = 1;
-
-		orderedArray.sort(function(a, b) {
-			var parsedA = parseInt(a);
-			var parsedB = parseInt(b);
-
-			if (isNaN(parsedA)) {
-				return -1;
-			} else if (isNaN(parsedB)) {
-				return 1;
-			} else {
-				return parsedA > parsedB ? 1 : -1;
-			}
-		});
+		var isLessThan = function(a, b) {
+			return _.every([
+				_.every([a, b], _.isNumber),
+				a < b
+			]);
+		};
+		var isNonConsecutive = function(number, index, list) {
+			var next = list[index + 1];
+			var uppderBound = _.isUndefined(next) ? Infinity : next;
```

Comment:
I think this is a small typo in the variable name:
```suggestion
			var upperBound = _.isUndefined(next) ? Infinity : next;
```